test/parallel/test-fs-read-stream.js

```diff
@@ -21,183 +21,199 @@
+  }));
 
-file.on('close', function() {
-  assert.strictEqual(bytesRead, fileSize);
-  assert.strictEqual(file.bytesRead, fileSize);
-  callbacks.close++;
+  process.on('exit', function() {
+    assert.strictEqual(file.length, 30000);
+  });
+}
 
-  //assert.strictEqual(fs.readFileSync(fn), fileContent);
-});
+{
+  const file = fs.createReadStream(fn, {encoding: 'utf8'});
+  file.length = 0;
+  file.on('data', function(data) {
+    assert.strictEqual('string', typeof data);
+    file.length += data.length;
+
+    for (let i = 0; i < data.length; i++) {
+      // http://www.fileformat.info/info/unicode/char/2026/index.htm
+      assert.strictEqual('\u2026', data[i]);
+    }
+  });
 
-const file3 = fs.createReadStream(fn, {encoding: 'utf8'});
-file3.length = 0;
-file3.on('data', function(data) {
-  assert.strictEqual('string', typeof data);
-  file3.length += data.length;
+  file.on('close', common.mustCall());
 
-  for (let i = 0; i < data.length; i++) {
-    // http://www.fileformat.info/info/unicode/char/2026/index.htm
-    assert.strictEqual('\u2026', data[i]);
-  }
-});
-
-file3.on('close', function() {
-  callbacks.close++;
-});
-
-process.on('exit', function() {
-  assert.strictEqual(1, callbacks.open);
-  assert.strictEqual(1, callbacks.end);
-  assert.strictEqual(2, callbacks.close);
-  assert.strictEqual(30000, file.length);
-  assert.strictEqual(10000, file3.length);
-  console.error('ok');
-});
-
-const file4 = fs.createReadStream(rangeFile, {bufferSize: 1, start: 1, end: 2});
-let contentRead = '';
-file4.on('data', function(data) {
-  contentRead += data.toString('utf-8');
-});
-file4.on('end', function(data) {
-  assert.strictEqual(contentRead, 'yz');
-});
-
-const file5 = fs.createReadStream(rangeFile, {bufferSize: 1, start: 1});
-file5.data = '';
-file5.on('data', function(data) {
-  file5.data += data.toString('utf-8');
-});
-file5.on('end', function() {
-  assert.strictEqual(file5.data, 'yz\n');
-});
-
-// https://github.com/joyent/node/issues/2320
-const file6 = fs.createReadStream(rangeFile, {bufferSize: 1.23, start: 1});
-file6.data = '';
-file6.on('data', function(data) {
-  file6.data += data.toString('utf-8');
-});
-file6.on('end', function() {
-  assert.strictEqual(file6.data, 'yz\n');
-});
+  process.on('exit', function() {
+    assert.strictEqual(file.length, 10000);
+  });
+}
+
+{
+  const file =
+    fs.createReadStream(rangeFile, {bufferSize: 1, start: 1, end: 2});
+  let contentRead = '';
+  file.on('data', function(data) {
+    contentRead += data.toString('utf-8');
+  });
+  file.on('end', function(data) {
+    assert.strictEqual(contentRead, 'yz');
+  });
+}
+
+{
+  const file = fs.createReadStream(rangeFile, {bufferSize: 1, start: 1});
+  file.data = '';
+  file.on('data', function(data) {
+    file.data += data.toString('utf-8');
+  });
+  file.on('end', function() {
+    assert.strictEqual(file.data, 'yz\n');
+  });
+}
+
+{
+  // https://github.com/joyent/node/issues/2320
```

Comment:
Will this live forever? Change to `// Ref: https://github.com/nodejs/node-v0.x-archive/issues/2320`?